presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -477,6 +477,20 @@ public void testAnalyzeFullTableDisallowed()
         assertUpdate(session, "DROP TABLE partition_test15");
     }
 
+    @Test
+    public void testIsNotNullWithNestedData()
+    {
+        Session admin = Session.builder(getQueryRunner().getDefaultSession())
+                .setIdentity(Identity.forUser("hive")
+                        .withRole("hive", new SelectedRole(ROLE, Optional.of("admin")))
+                        .build())
+                .build();
+        assertUpdate(admin, "create table nest_test(id int, a row(x varchar, y timestamp, z varchar), b varchar) WITH (format='PARQUET')");
+        assertUpdate(admin, "insert into nest_test values(0, null, '1')", 1);
+        computeActual(admin, "select id from nest_test where a.y is not null");
```

Comment:
The issue is that when a.y is read, the parquet reader doesn't think that the value is null because of this check in ParquetTypeUtils:

    public static boolean isValueNull(boolean required, int definitionLevel, int maxDefinitionLevel)
    {
        return !required && (definitionLevel == maxDefinitionLevel - 1);
    }

If it's changed to `definitionLevel == maxDefinitionLevel - 1`, then it works for this test, but breaks others.

In the end, the problem stems from positionCount in LongArrayBlockBuilder being 0 here while it should be 1:

    @Override
    public Block build()
    {
        if (!hasNonNullValue) {
            return new RunLengthEncodedBlock(NULL_VALUE_BLOCK, positionCount);
        }
        return new LongArrayBlock(0, positionCount, hasNullValue ? valueIsNull : null, values);
    }

Note that with ORC, the value of positionCount is 1 here.